### ISSUE TITLE
[simd] Build error on arm64-linux

### ISSUE THREAD

**Issue by TonyCongqianWang** [microsoft/vcpkg#36496]
Hello,
I got a fresh installation of rasberry os lite and cloned and bootstrapped vcpkg. I was able to install another package, but when I tried to install simd, I receive a build error. The logs say something about permission denied. Running it with sudo didn't work because of a missing enviroment variable

**Comment by TonyCongqianWang** [comment]
I have installed the newest version of cmake ninja and vcpkg and the recommended packages build-essential tar curl zip unzip.

**Comment by TonyCongqianWang** [comment]
With sudo -E I still get the same error log:

[1/305] "" -DSIMD_INT8_DEBUG_DISABLE -I/home/pi/packages/vcpkg/buildtrees/simd/src/v5.3.128-d09ac2fee4.clean/prj/cmake/../../src -fPIC -g -fPIC -fPIC -std=c++11 -O0 -g -fm>
FAILED: CMakeFiles/Simd.dir/home/pi/packages/vcpkg/buildtrees/simd/src/v5.3.128-d09ac2fee4.clean/src/Simd/SimdLib.cpp.o
"" -DSIMD_INT8_DEBUG_DISABLE -I/home/pi/packages/vcpkg/buildtrees/simd/src/v5.3.128-d09ac2fee4.clean/prj/cmake/../../src -fPIC -g -fPIC -fPIC -std=c++11 -O0 -g -fmax-error>
/bin/sh: 1: : Permission denied
[2/305] "" -DSIMD_INT8_DEBUG_DISABLE -I/home/pi/packages/vcpkg/buildtrees/simd/src/v5.3.128-d09ac2fee4.clean/prj/cmake/../../src -fPIC -g -fPIC -fPIC -std=c++11 -O0 -g -fm>
FAILED: CMakeFiles/Simd.dir/home/pi/packages/vcpkg/buildtrees/simd/src/v5.3.128-d09ac2fee4.clean/src/Simd/SimdBaseAbsDifference.cpp.o
"" -DSIMD_INT8_DEBUG_DISABLE -I/home/pi/packages/vcpkg/buildtrees/simd/src/v5.3.128-d09ac2fee4.clean/prj/cmake/../../src -fPIC -g -fPIC -fPIC -std=c++11 -O0 -g -fmax-error>
/bin/sh: 1: : Permission denied
[3/305] "" -DSIMD_INT8_DEBUG_DISABLE -I/home/pi/packages/vcpkg/buildtrees/simd/src/v5.3.128-d09ac2fee4.clean/prj/cmake/../../src -fPIC -g -fPIC -fPIC -std=c++11 -O0 -g -fm>
FAILED: CMakeFiles/Simd.dir/home/pi/packages/vcpkg/buildtrees/simd/src/v5.3.128-d09ac2fee4.clean/src/Simd/SimdBaseAbsDifferenceSum.cpp.o
"" -DSIMD_INT8_DEBUG_DISABLE -I/home/pi/packages/vcpkg/buildtrees/simd/src/v5.3.128-d09ac2fee4.clean/prj/cmake/../../src -fPIC -g -fPIC -fPIC -std=c++11 -O0 -g -fmax-error>
/bin/sh: 1: : Permission denied
[4/305] "" -DSIMD_INT8_DEBUG_DISABLE -I/home/pi/packages/vcpkg/buildtrees/simd/src/v5.3.128-d09ac2fee4.clean/prj/cmake/../../src -fPIC -g -fPIC -fPIC -std=c++11 -O0 -g -fm>
FAILED: CMakeFiles/Simd.dir/home/pi/packages/vcpkg/buildtrees/simd/src/v5.3.128-d09ac2fee4.clean/src/Simd/SimdBaseAddFeatureDifference.cpp.o
"" -DSIMD_INT8_DEBUG_DISABLE -I/home/pi/packages/vcpkg/buildtrees/simd/src/v5.3.128-d09ac2fee4.clean/prj/cmake/../../src -fPIC -g -fPIC -fPIC -std=c++11 -O0 -g -fmax-error>
/bin/sh: 1: : Permission denied
[5/305] "" -DSIMD_INT8_DEBUG_DISABLE -I/home/pi/packages/vcpkg/buildtrees/simd/src/v5.3.128-d09ac2fee4.clean/prj/cmake/../../src -fPIC -g -fPIC -fPIC -std=c++11 -O0 -g -fm>
FAILED: CMakeFiles/Simd.dir/home/pi/packages/vcpkg/buildtrees/simd/src/v5.3.128-d09ac2fee4.clean/src/Simd/SimdBaseAbsGradientSaturatedSum.cpp.o
"" -DSIMD_INT8_DEBUG_DISABLE -I/home/pi/packages/vcpkg/buildtrees/simd/src/v5.3.128-d09ac2fee4.clean/prj/cmake/../../src -fPIC -g -fPIC -fPIC -std=c++11 -O0 -g -fmax-error>
/bin/sh: 1: : Permission denied
ninja: build stopped: subcommand failed.

**Comment by dg0yt** [comment]
`sudo` is not the answer, and
`""` is not an executable program.

**Comment by StarGate-One** [comment]
See #30604 How to properly report a package build failure.

**Comment by TonyCongqianWang** [comment]
I tried both
    sudo -E ./vcpkg install simd
and
    ./vcpkg install simd
On the newest version and both did not work

I managed to build it using an older version (git hash a42af01b7) with 
    sudo -E ./vcpkg install simd
Sudo was in fact the answer here. Without sudo it did not build on that version either.

**Comment by dg0yt** [comment]
> Sudo was in fact the answer here. Without sudo it did not build on that version either.

No. `sudo` is part of the problem. You are creating files with owner `root` when they should be owned by your user account.
And then you cannot modify them without `sudo` next time.
No to mention that broken or evil build scripts might overwrite everything while running with `sudo`.
So the system is already broken.

You need to find why the compiler isn't identified correctly. In your user's context. Check the environment variables. There must be a reason why the builds picks the empty string. If it is displayed correctly here, given the absence of markup.

**Comment by TonyCongqianWang** [comment]
I am aware, that the scripts are supposed to function without using sudo, but in this case it didn't. Thats why I created an issue. It did work with sudo in the older commit but it should not require sudo.

I gave you all the information I have and all the Information that is needed to recreate the behaviour. I used a fresh install of Raspberry OS lite (newest version to that date). Pulled the newest version (same date) of vcpkg, used the bootstrap script and tried to install simd - did not work. I don't know why the compiler was not identified correctly thats again, part of the reason why I created the issue

**Comment by dg0yt** [comment]
> I gave you all the information I have.

- the result of checking your environment variables.
- the config logs created in `buildtrees/simd`.
